### PR TITLE
Add question responses to teacher application detail view

### PR DESCRIPTION
**Pull request by megcrenshaw**
Adds four question responses to the detail view of the teacher application:
- 'What is your current role at your school?'
- 'Have you participated in previous yearlong Code.org Professional Learning Programs? If so, mark the programs you've participated in.'
- 'Have you previously taught CS or have you learned CS yourself?'
-  'Are you able to independently write a function (or procedure) with one or more parameters and that uses conditional logic, loops, and an array (or list)?'

The "meets guidelines" portion [is defined in teacher_application.rb](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/application/teacher_application.rb#L989) –– in particular:
- the application DOES meet guidelines if the applicant has never taken PD *of this exact course* before. It does NOT meet guidelines if they have taken this course's PD before.
- the application DOES NOT meet guidelines if the applicant answered "no" to either of the CSA screening questions

Tested this locally:
image

There are currently no unit tests to ensure individual questions are being shown, and I'm not sure it makes sense to have unit tests, as they're coming through via ruby-defined constants and there's an eyes test that will capture changes. Happy to have more discussion here though.

## Links

See https://codedotorg.atlassian.net/browse/ACQ-350 for link to discussion about this –– we're showing all demographic information, minus race and gender, in a new section.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx [hunk 1]
```diff
@@ -720,69 +720,65 @@ export class DetailViewContents extends React.Component {
...
-                      )
-                    );
-                  })}
-                </tbody>
-              </Table>
-            </div>
-          )
-        )}
+        {Object.keys(this.sectionHeaders).map((header, i) => (
+          <div key={i}>
```
Comment (megcrenshaw) [inline apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx:728]: This chunk remains untouched –– removed the `sectionsToRemove` being pulled out

**Comment by dmcavoy** [comment]
For the two CSA specific questions do those only show if someone has answered them? (Or if they have a CSA application?)

**Comment by megcrenshaw** [comment]
> For the two CSA specific questions do those only show if someone has answered them? (Or if they have a CSA application?)

Good question. Yes, for CSD, it looks like this
image

**Comment by megcrenshaw** [comment]
Just went on quite a scavenger hunt to figure out how those weren't showing. If an applicant answers CSA-related questions, they *do* get saved in the form data, and admins can see that in the detail view via "Edit Form Data." But when we go to show the form data in the "basic" detail view (i.e. not the raw form data), we filter them out here https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/application/application_base.rb#L272, and the filtered labels are defined in the TeacherApplication model https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/application/teacher_application.rb#L790-L834.

Now I can sleep at night 😴

**Review by davidsbailey — APPROVED** [review]

**Comment by davidsbailey** [comment]
nice sleuthing Meg!

**Pull request merged**